js/repeater-list.js

```diff
@@ -881,79 +826,69 @@
 				columns.push(actionsColumn);
 			}
 
-			this.list_columns = columns;
 
-			var $headerRow = $('<tr></tr>');
-			for (var i = 0; i < columns.length; i++) {
-				renderHeader.call(this, $headerRow, columns[i], i);
+			$thead = $('<thead data-preserve="deep"><tr></tr></thead>');
+			$tr = $thead.find('tr');
+			for (i = 0, l = columns.length; i < l; i++) {
+				renderHeader.call(this, $tr, columns, i);
 			}
-
-			$thead = $('<thead data-preserve="deep"></thead>');
-			$thead.append($headerRow);
 			$table.prepend($thead);
 
-			// after checkbox column is created need to get width of checkbox column from its css class
 			if (this.viewOptions.list_selectable === 'multi' && !this.list_noItems) {
+				//after checkbox column is created need to get width of checkbox column from
+				//its css class
 				var checkboxWidth = this.$element.find('.repeater-list-wrapper .header-checkbox').outerWidth();
-				columns[0].width = checkboxWidth;
+				var selectColumn = $.grep(columns, function(column){
+					return column.property === '@_CHECKBOX_@';
+				})[0];
+				selectColumn.width = checkboxWidth;
 			}
-
-			sizeColumns.call(this, $headerRow);
+			sizeColumns.call(this, $tr);
 		}
-	};
+	}
 
-	var sizeColumns = function sizeColumns ($tr) {
-		var autoGauge = [];
+	function sizeColumns ($tr) {
+		var auto = [];
```

Comment:
This variable name needs to be clearer, "auto" is too short and does not describe what it holds and what it is used for.